### PR TITLE
HTML5 support

### PR DESCRIPTION
**Pull request by sergeysinyavsky**
Added HTML5 support using [navigator.share()](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/share)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,16 +25,19 @@ Share text using a native sharing dialog.
...
+### share.image(bytes, [text], [file_name])
+Share an image (with optional text) using a native sharing dialog. The image format must be supported by ```BitmapFactory.decodeByteArray()``` on Android, ```UIImage.initWithData``` on iOS/OSX and [DataURI](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URIs) on HTML5. On Android the image will first be stored locally as a file and then shared using a FileProvider. In order for this to work you need to follow the additional setup steps for Android mentioned above.
 
 **PARAMETERS**
 * ``` bytes``` (string) - The image bytes to share
 * ```text``` (string) - Optional text to share
+* ```file_name``` (string) - Optional Only for `HTML5` image name. Default `"file.png"`
 
-### share.file(path, [text])
+### share.file(path, [text], [file_name])
```
Comment (britzl) [inline README.md:36]: I'm thinking about argument order and how to make HTML5 more closely resemble  the other platforms. With your suggestion both path and text are given a different meaning in html5

Wouldn't this be better?

```
-- html5
-- third argument is a table with options (can be used for extra params on all platforms)
share.file("file.txt", "file contents", { type = "text/plain" }

-- others
share.file("path/to/file.txt", "extra text")
```

**Review by britzl — CHANGES_REQUESTED** [review]
Thank you for the contribution. I have some feedback on argument order for share.file() that I think we should discuss.

File: README.md (hunk 1 above)
Comment (sergeysinyavsky) [reply README.md]: @britzl I agree. Changed parameters for html version

File: README.md [hunk 2]
```diff
@@ -24,16 +25,25 @@ Share text using a native sharing dialog.
...
+### share.file(path, [text], [data])
 Share a file (with optional text) using a native sharing dialog. On Android the file will first be copied to a predefined location and then shared using a FileProvider. In order for this to work you need to follow the additional setup steps for Android mentioned above. Files are shared with their original filename and extension. This will allow iOS to offer different kinds of applications depending on the shared content.
 
 **PARAMETERS**
-* ``` path``` (string) - Full path to the file to share
-* ```text``` (string) - Optional text to share
+* ``` path``` (string) - Full path to the file to share. For `HTML5`: specify the name of the file. Default `file.txt`.
+* ```text``` (string) - Optional text to share. For `HTML5`: specify the data that should be in the file.
+
+* ```data``` (table) - Optional Only for `HTML5`.
```
Comment (britzl) [inline README.md:43]: It's not data is it? More like config values or options. We usually call this table `options` (see http.request for instance)

**Review by britzl — CHANGES_REQUESTED** [review]
Great stuff! One final thing with about data vs options

File: README.md [hunk 3]
```diff
@@ -24,16 +25,24 @@ Share text using a native sharing dialog.
...
+### share.image(bytes, [text], [file_name])
+Share an image (with optional text) using a native sharing dialog. The image format must be supported by ```BitmapFactory.decodeByteArray()``` on Android, ```UIImage.initWithData``` on iOS/OSX and [DataURI](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URIs) on HTML5. On Android the image will first be stored locally as a file and then shared using a FileProvider. In order for this to work you need to follow the additional setup steps for Android mentioned above.
 
 **PARAMETERS**
 * ``` bytes``` (string) - The image bytes to share
 * ```text``` (string) - Optional text to share
+* ```file_name``` (string) - Optional Only for `HTML5` image name. Default `"file.png"`
 
-### share.file(path, [text])
+### share.file(path, [text], [data])
```
Comment (britzl) [inline README.md:36]: data -> options

**Review by britzl — CHANGES_REQUESTED** [review]
One last thing!

**Review by britzl — APPROVED** [review]
Thank you!

**Pull request merged**